docs/Stream-Protocol-Spec.md

```diff
@@ -0,0 +1,88 @@
+
+ 
+stream termination condition:
+ - timeout, connection died, we get an error and remove the client, server also gets an error from p2p layer and removes all servers/clients and drops the peer
+
+considerations:
+- server must make sure that chunk got to client in order to account in SWAP (synchronous). if the send does not result in an error - the send should be accounted
+- there is always a max batch size so that clients cannot grieve servers with very large ranges
+
+syncing contracts:
+ - stream indexes always > 0
+ - syncing is an implementation of the stream protocol
+ - client is expected to manage all intervals, and therefore:
+ - server is designed to be stateless, except for the case of managing a offered/wanted roundtrip and the knowledge of a boundedness of a stream (e.g. the server knows that syncing streams are always unbounded from the localstore perspective - data can always enter the system, however this is not the case for live video stream for example)
+ - the server does not terminate streams - it is at the discretion of the downstream peer
+ - the server does not initiate any messages unless instructed to
+ - the server does not instruct client on which bins to subscribe to it
+
+
+| Msg Name | From->To | Params   | Example |
+| -------- | -------- | -------- | ------- |
+| Stream   | Client->Server  | Streams`[]string` | `SYNC\|6, SYNC\|5` |
+| StreamAck   | Server->Client  | Streams`[]Info` <br>Stream`string`<br>SessionIdx`uint64` <br>Bounded`bool` | `SYNC\|6;CUR=1632;bounded, SYNC\|7;CUR=18433;bounded` |
+| GetRange | Client->Server| Ruid`uint`<br>Stream `string`<br>From`uint`<br>To`*uint`(nullable) | `Stream: SYNC\|6, From: 1, To: 100`(bounded)<br>`Stream: SYNC\|7, From: 109`(unbounded) | 
+| OfferedHashes | Server->Client| Ruid`uint`<br>Hashes `[]byte` | `Hashes: [cbcbbaddda, bcbbbdbbdc, ....]` |
+| WantedHashes | Client->Server | Ruid`uint`<br>Bitvector`[]byte` | `Bitvector: [0100100100] ` |
+| ChunkDelivery | Server->Client | Chunk `[]byte` | `Stream: SYNC\|6, Chunk: [001000101]` |
+| BatchDone | Server->Client| Ruid `uint`<br>Last `uint` | `Ruid: 21321, Last: 113331` |
+| StreamState | Client<->Server | Stream`string`<br>Code`uint16`<br>Message`string`| `Stream: SYNC\|6, Code:1, Message:"Stream became bounded"`<br>`Stream: SYNC\|5, Code:2, Message: "No such stream"` |
+
+
+Notes:
+* communicating the last bin index when roundtrip is configured - can be done on top of OfferedHashes message (alongside the hashes), or to reuse the ACK from the no-roundtrip config
+* two notions of bounded - on the stream level and on the localstore
+* if TO is not specified - we assume unbounded stream, and we just send whatever, until at most, we fill up an entire batch.
+
+
+Message exchange examples:
+
+Initial handshake - client queries server for stream states
+![handshake](https://raw.githubusercontent.com/ethersphere/swarm/stream-spec/stream-spec/docs/diagrams/stream-handshake.png)
```

Comment:
The image link is wrong. Should be `https://raw.githubusercontent.com/ethersphere/swarm/stream-spec/stream-spec/docs/diagrams/stream-handshake.png` 

Also, maybe you can just use relative paths for this? e.g. `diagrams/stream-handshake.png`

Reply:
> Also, maybe you can just use relative paths for this? e.g. diagrams/stream-handshake.png

 unfortunately that won't work since it brings the assets together will the whole github html packaging. when you use `raw.githubusercontent.com` you get just the asset itself